recipe/meta.yaml

```diff
@@ -30,9 +30,6 @@ package:
 source:
   url: https://github.com/ornladios/ADIOS2/archive/refs/tags/v{{ version }}.tar.gz
   sha256: {{ sha256 }}
-  patches:
-    - python-exclude-from-all.patch
```

Comment:
Oh, this patch we need to keep and update.

Reply:
I figure out that we can get around this by specifying components in the install invokation. I enabled our python bindings and library for the libadios2. The reason of it is that in 2.10 we have substantially extended it. Let me know if this is problematic.

Reply:
hmm, nevermind including python binding/API in the libadios2, this is frown upon in the conda documentation. They actually recommend the following scheme of three packages:

- libadios2
- py-adios2
- adios2

This might be overkill and something more for a future discussion. AS for now I have reverted some of my changes to have python files and scripts only be installed in the libadios2 pkg.

Reply:
Thanks!

Just CCing: in the past, @minrk helped us to introduce the split here :)

Reply:
Yeah, if this is supported upstream without a patch, that's great! I just _really_ wish [cmake could report a list of every component](https://gitlab.kitware.com/cmake/cmake/-/issues/24335) or otherwise make a way to install "everything _but_ component X", because I don't know how to feasibly maintain the component list for future releases. If anyone knows a good way to collect component lists, that would be super handy.

If you want to split `adios2` and `py-adios2`, that's okay with me, but I'm not sure it's really needed. The only really important thing is that `libadios2` doesn't have a Python dependency and `adios2` has a full standard install. I'm not sure the difference between `py-adios2` and `adios2` is worth the maintenance.

Reply:
I just learned you can `make list_install_components` to see the components, so that helps. It produces:

```
Available install components are: "Unspecified" "adios2_atl-development" "adios2_atl-libraries" "adios2_c-development" "adios2_c-libraries" "adios2_core-config" "adios2_core-development" "adios2_core-libraries" "adios2_cxx11-development" "adios2_cxx11-libraries" "adios2_dill-development" "adios2_dill-libraries" "adios2_enet-development" "adios2_enet-libraries" "adios2_evpath-development" "adios2_evpath-libraries" "adios2_ffs-development" "adios2_ffs-libraries" "adios2_scripts-runtime" "adios2_tools-data" "adios2_tools-runtime" "cmenet" "cmmulticast" "cmselect" "cmsockets" "cmudp"
```

 There doesn't appear to be any _cmake_ command to produce this info, but at least it's somewhere.

We could try to be fancy and do some parsing of `make list_install_components` to implement "install everything BUT X", but manually checking this list on release is probably going to be good enough.